pkg/dockerregistry/server/auth.go

```diff
@@ -201,6 +196,11 @@ func TokenRealm(options map[string]interface{}) (*url.URL, error) {
 func newAccessController(options map[string]interface{}) (registryauth.AccessController, error) {
 	log.Info("Using Origin Auth handler")
 
+	ctx, ok := options["_context"].(context.Context)
```

Comment:
I should've replace log.Info with context.GetLogger(ctx), but I forged to do that. And I think it's convenient to have the application context in all hooks, not only in the middlewares.